### PR TITLE
Ocular machine config

### PR DESCRIPTION
**Pull request by NWeber-SL**

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
   * [ShiftLeft Requirements](introduction/requirements.md)
 * [Using ShiftLeft Ocular](using-ocular/ocular-desc.md)
   * [Quick Start](using-ocular/ocular-quick-start.md)
+  * [Performance Optimization](using-ocular/ocular-config.md)
```
Comment (vlad-alexandru-ionescu) [inline SUMMARY.md]: The name of the file is too generic. We should use something more specific.

File: using-ocular/ocular-config.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# ShiftLeft Ocular Performance Optimization for Analyzing Java Applications
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: This applies to any language, not just Java. I would call this page

> ShiftLeft Ocular Memory Size Recommendations

to make it more suggestive of the information in it.

File: using-ocular/ocular-config.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+By default, disk overflow is enabled for Code Property Graphs (CPG). Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other secitions of the CPG are swapped onto disk.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: It's not clear what "disk overflow" is. We should perhaps link to another page where this is described.

Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: secitions -> sections

Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: I would avoid the word "swapped" as it has a very specific meaning in linux-based systems when it comes to memory. We're not using the swap space for this feature.

Instead we should say "temporarily stored to disk" or something along these lines.

File: using-ocular/ocular-config.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# ShiftLeft Ocular Performance Optimization for Analyzing Java Applications
+
+By default, disk overflow is enabled for Code Property Graphs (CPG). Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other secitions of the CPG are swapped onto disk. 
+
+However, when analyzing large Java Applications, disk overflow can slow down performance. Instead, you can optimize ShiftLeft Ocular performance through the appropriate combination of memory and disks sizes.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: Not just Java - any language.

File: using-ocular/ocular-config.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# ShiftLeft Ocular Performance Optimization for Analyzing Java Applications
+
+By default, disk overflow is enabled for Code Property Graphs (CPG). Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other secitions of the CPG are swapped onto disk. 
+
+However, when analyzing large Java Applications, disk overflow can slow down performance. Instead, you can optimize ShiftLeft Ocular performance through the appropriate combination of memory and disks sizes. 
+
+Note that estimating the necessary amount of memory is non-trivial; the size of the CPG on disk is the best predictor. ShiftLeft Ocular determines the amount of memory need and presents the following (example) information when you load or [create a CPG](../tutorials/cpgcreate.md):
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]: I'd say this part is an implementation detail that the user does not need to know: "the size of the CPG on disk is the best predictor"

File: using-ocular/ocular-config.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+createCpg("someBigApp.jar")
+// ...
+// debug output from java2cpg
+// ...
+The cpg.bin.zip you're trying to load is 63MB on disk and (based on a very rough calculation) will require ~12712MB heap, but the current max heap size is 455MB.
+We suggest to provide a large enough `-J-Xmx` setting, e.g. `ocular.sh -J-Xmx14g`
+Please make sure that your machine has sufficient free physical memory available.
+If you choose to disregard this message, the loading of the cpg might take longer, as the disk will be used to store part of the graph.
+```
+You can cancel the loading of the current CPG using Ctrl-c, close ShiftLeft Ocular, and then restart it with more memory, e.g. ./ocular.sh -J-Xmx14g for 14GB heap.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]:
```suggestion
You can cancel the loading of the current CPG using `Ctrl-c`, close ShiftLeft Ocular, and then restart it with more memory, e.g. `./ocular.sh -J-Xmx14g` for 14GB heap.
```

File: SUMMARY.md (hunk 1 above)
Comment (NWeber-SL) [reply SUMMARY.md]: Using your suggestion "ShiftLeft Ocular Memory Size Recommendations"

File: using-ocular/ocular-config.md (hunk 2 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Done

File: using-ocular/ocular-config.md (hunk 3 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Linking to description in quick start

Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Fixed.

Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Done

File: using-ocular/ocular-config.md (hunk 4 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Fixed

File: using-ocular/ocular-config.md (hunk 5 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Removed

File: using-ocular/ocular-config.md (hunk 6 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Done

File: using-ocular/ocular-config.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# ShiftLeft Ocular Memory Size Recommendations
+
+By default, disk overflow is enabled for Code Property Graphs (CPG). Disk overflow moves cache entries to disk only when the number of entries in memory exceed the maximum memory allocation. Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other sections of the CPG are temporarily stored to disk. 
+
+However, when analyzing large applications, disk overflow can slow down performance. Instead, you can optimize ShiftLeft Ocular performance through the appropriate combination of memory and disks sizes.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]:
```suggestion
However, when analyzing large applications, disk overflow can slow down performance. Instead, you can optimize ShiftLeft Ocular performance through the appropriate combination of memory and disk sizes. 
```

File: using-ocular/ocular-config.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# ShiftLeft Ocular Memory Size Recommendations
+
+By default, disk overflow is enabled for Code Property Graphs (CPG). Disk overflow moves cache entries to disk only when the number of entries in memory exceed the maximum memory allocation. Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other sections of the CPG are temporarily stored to disk. 
+
+However, when analyzing large applications, disk overflow can slow down performance. Instead, you can optimize ShiftLeft Ocular performance through the appropriate combination of memory and disks sizes. 
+
+Note that estimating the necessary amount of memory is non-trivial. ShiftLeft Ocular determines the amount of memory need and presents the following (example) information when you load or [create a CPG](../tutorials/cpgcreate.md):
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-config.md]:
```suggestion
Note that estimating the necessary amount of memory is non-trivial. ShiftLeft Ocular determines the amount of memory needed and presents the following (example) information when you load or [create a CPG](../tutorials/cpgcreate.md): 
```

File: SUMMARY.md [hunk 9]
```diff
@@ -8,6 +8,7 @@
   * [ShiftLeft Requirements](introduction/requirements.md)
 * [Using ShiftLeft Ocular](using-ocular/ocular-desc.md)
   * [Quick Start](using-ocular/ocular-quick-start.md)
+  * [Memory Size Recommendations](using-ocular/ocular-config.md)
```
Comment (vlad-alexandru-ionescu) [inline SUMMARY.md]: I also meant the file name itself. `ocular-config.md` is too generic as there are many types of configuration. Perhaps should be renamed to `ocular-memory-size.md`

File: using-ocular/ocular-config.md (hunk 7 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Done

File: using-ocular/ocular-config.md (hunk 8 above)
Comment (NWeber-SL) [reply using-ocular/ocular-config.md]: Done

File: SUMMARY.md (hunk 9 above)
Comment (NWeber-SL) [reply SUMMARY.md]: Done

**Pull request merged**